src/dynamicprompts/generators/randomprompt.py

```diff
@@ -68,7 +68,8 @@ def generate(
             prompts = []
             for seed in seeds:
                 self._context.rand.seed(seed)
-                prompts.append(str(next(iter(gen))))
+                if gen:
+                    prompts.append(str(next(iter(gen))))
```

Comment:
Why bother looking at `if gen:` _inside_ the loop? If there are 10,000 seeds, this will end up just seeding the generator 10,000 times and do nothing with it.

It looks like `gen = self._context.sample_prompts(template, num_images)` should be before `if seeds:`, and `if not gen: return []` after it.

The non-`seeds` branch should then also use the same `gen`: `return [str(p) for p in gen]`.